crates/astria-composer/src/executor/builder.rs

```diff
@@ -102,13 +108,12 @@ fn read_signing_key_from_file<P: AsRef<Path>>(path: P) -> eyre::Result<SigningKe
     Ok(SigningKey::from(private_key_bytes))
 }
 
-fn connect_sequencer_grpc(
-    grpc_endpoint: &str,
-) -> eyre::Result<SequencerServiceClient<tonic::transport::Channel>> {
-    let uri: tonic::transport::Uri = grpc_endpoint
+fn connect_sequencer_grpc(grpc_endpoint: &str) -> eyre::Result<SequencerServiceClient<Channel>> {
+    let uri: Uri = grpc_endpoint
         .parse()
         .wrap_err("failed to parse endpoint as URI")?;
-    Ok(SequencerServiceClient::new(
-        tonic::transport::Endpoint::from(uri).connect_lazy(),
-    ))
+    let endpoint = Endpoint::from(uri.clone())
```

Comment:
Unclear why the clone here is necessary. Just move the URI in there?

Alternatively remove the `let uri` binding altogether and chain the whole thing:

```suggestion
    let endpoint = grpc_endpoint.parse::<Endpoint>().wrap_err("failed to parse input as gRPC endpoint")?.tls_config(...) // etc
```